### ISSUE TITLE
why loss is vary big when i trained about 8000 iters? And , can't detect object.

### ISSUE THREAD

**Issue by liwanning** [eric612/MobileNet-YOLO#136]
I0724 11:01:42.380015 11724 sgd_solver.cpp:121] Iteration 8330, lr = 1e-05
I0724 11:01:43.098712 11724 yolov3_layer.cpp:411] noobj: 0.003987 obj: 0.0249371 iou: 0.631845 cat: 0.743798 recall: 0.840907 recall75: 0.212433 count: 10
I0724 11:01:43.185757 11724 yolov3_layer.cpp:411] noobj: 0.00150616 obj: 0.0162007 iou: 0.4047 cat: 0.740208 recall: 0.380869 recall75: 0.0275666 count: 20
I0724 11:01:51.990276 11724 yolov3_layer.cpp:411] noobj: 0.00276174 obj: 0.0129962 iou: 0.593776 cat: 0.775496 recall: 0.756778 recall75: 0.102167 count: 11
I0724 11:01:52.038159 11724 yolov3_layer.cpp:411] noobj: 0.000737336 obj: 0.0180812 iou: 0.451522 cat: 0.724751 recall: 0.418821 recall75: 0.00682012 count: 14
I0724 11:01:56.291533 11724 yolov3_layer.cpp:411] noobj: 0.00454227 obj: 0.0303835 iou: 0.638303 cat: 0.715216 recall: 0.871104 recall75: 0.240916 count: 10
I0724 11:01:56.318351 11724 yolov3_layer.cpp:411] noobj: 0.00207978 obj: 0.0212043 iou: 0.371831 cat: 0.650692 recall: 0.329464 recall75: 0.0417203 count: 18
I0724 11:01:59.938424 11724 yolov3_layer.cpp:411] noobj: 0.00474432 obj: 0.0361005 iou: 0.631767 cat: 0.805605 recall: 0.809413 recall75: 0.214743 count: 9
I0724 11:01:59.947917 11724 yolov3_layer.cpp:411] noobj: 0.00250914 obj: 0.0207157 iou: 0.447481 cat: 0.757216 recall: 0.46424 recall75: 0.0213416 count: 20
I0724 11:02:06.675048 11724 yolov3_layer.cpp:411] noobj: 0.00342628 obj: 0.0167354 iou: 0.62796 cat: 0.796089 recall: 0.802746 recall75: 0.198857 count: 11
I0724 11:02:06.715499 11724 yolov3_layer.cpp:411] noobj: 0.00100398 obj: 0.0103554 iou: 0.439788 cat: 0.703969 recall: 0.412263 recall75: 0.0273504 count: 15
I0724 11:02:10.893319 11724 yolov3_layer.cpp:411] noobj: 0.00470649 obj: 0.0333528 iou: 0.652693 cat: 0.796599 recall: 0.861371 recall75: 0.219222 count: 10
I0724 11:02:10.924726 11724 yolov3_layer.cpp:411] noobj: 0.00234764 obj: 0.0138397 iou: 0.399392 cat: 0.678624 recall: 0.318984 recall75: 0.00735043 count: 15
I0724 11:02:19.380259 11724 yolov3_layer.cpp:411] noobj: 0.00302152 obj: 0.018018 iou: 0.619263 cat: 0.760267 recall: 0.792087 recall75: 0.201848 count: 12
I0724 11:02:19.428537 11724 yolov3_layer.cpp:411] noobj: 0.000691867 obj: 0.00703582 iou: 0.453972 cat: 0.710493 recall: 0.433218 recall75: 0.026619 count: 14
I0724 11:02:21.289041 11724 solver.cpp:252] Iteration 8340 (0.257012 iter/s, 38.9086s/10 iters), loss = 10.1094
I0724 11:02:21.289105 11724 solver.cpp:271]     Train net output #0: det_loss1 = 4.67092 (* 1 = 4.67092 loss)
I0724 11:02:21.289113 11724 solver.cpp:271]     Train net output #1: det_loss2 = 4.01937 (* 1 = 4.01937 loss)
I0724 11:02:21.289119 11724 sgd_solver.cpp:121] Iteration 8340, lr = 1e-05
I0724 11:02:23.998903 11724 yolov3_layer.cpp:411] noobj: 0.0046585 obj: 0.0272238 iou: 0.631206 cat: 0.779213 recall: 0.835007 recall75: 0.223597 count: 9
I0724 11:02:24.015974 11724 yolov3_layer.cpp:411] noobj: 0.00235912 obj: 0.0204141 iou: 0.412968 cat: 0.698261 recall: 0.38797 recall75: 0.0297306 count: 16
I0724 11:02:31.552971 11724 yolov3_layer.cpp:411] noobj: 0.00323086 obj: 0.0176795 iou: 0.611877 cat: 0.82164 recall: 0.762585 recall75: 0.216875 count: 11
I0724 11:02:31.594913 11724 yolov3_layer.cpp:411] noobj: 0.00109292 obj: 0.0148009 iou: 0.40679 cat: 0.767 recall: 0.336258 recall75: 0.028449 count: 15
I0724 11:02:36.537789 11724 yolov3_layer.cpp:411] noobj: 0.00433756 obj: 0.0295993 iou: 0.664431 cat: 0.777312 recall: 0.924517 recall75: 0.209801 count: 10
I0724 11:02:36.563627 11724 yolov3_layer.cpp:411] noobj: 0.00170623 obj: 0.0146573 iou: 0.396563 cat: 0.718357 recall: 0.317458 recall75: 0.0106066 count: 18
I0724 11:02:44.751507 11724 yolov3_layer.cpp:411] noobj: 0.00293659 obj: 0.0182758 iou: 0.623124 cat: 0.783088 recall: 0.871752 recall75: 0.128711 count: 10
I0724 11:02:44.805984 11724 yolov3_layer.cpp:411] noobj: 0.000837113 obj: 0.0125956 iou: 0.45266 cat: 0.741478 recall: 0.436758 recall75: 0.0893333 count: 18
I0724 11:02:52.647768 11724 yolov3_layer.cpp:411] noobj: 0.00318034 obj: 0.0152355 iou: 0.640522 cat: 0.783716 recall: 0.83738 recall75: 0.235664 count: 11
I0724 11:02:52.663350 11724 yolov3_layer.cpp:411] noobj: 0.000894482 obj: 0.0128753 iou: 0.460856 cat: 0.781283 recall: 0.478207 recall75: 0.0511017 count: 16
I0724 11:02:59.085646 11724 yolov3_layer.cpp:411] noobj: 0.00360952 obj: 0.0186712 iou: 0.644686 cat: 0.765699 recall: 0.87178 recall75: 0.226435 count: 12
I0724 11:02:59.109361 11724 yolov3_layer.cpp:411] noobj: 0.0012199 obj: 0.0162013 iou: 0.429684 cat: 0.75107 recall: 0.387594 recall75: 0.0448355 count: 15
I0724 11:03:00.709214 11724 solver.cpp:252] Iteration 8350 (0.25368 iter/s, 39.4197s/10 iters), loss = 9.7169
I0724 11:03:00.709260 11724 solver.cpp:271]     Train net output #0: det_loss1 = 3.68264 (* 1 = 3.68264 loss)
I0724 11:03:00.709283 11724 solver.cpp:271]     Train net output #1: det_loss2 = 8.19198 (* 1 = 8.19198 loss)
I0724 11:03:00.709290 11724 sgd_solver.cpp:121] Iteration 8350, lr = 1e-05
I0724 11:03:03.196969 11724 yolov3_layer.cpp:411] noobj: 0.00440191 obj: 0.0282689 iou: 0.664532 cat: 0.771532 recall: 0.908641 recall75: 0.269627 count: 9
I0724 11:03:03.221573 11724 yolov3_layer.cpp:411] noobj: 0.00168547 obj: 0.0173269 iou: 0.430735 cat: 0.728044 recall: 0.44704 recall75: 0.0142857 count: 13
I0724 11:03:10.905336 11724 yolov3_layer.cpp:411] noobj: 0.00312919 obj: 0.0167264 iou: 0.635436 cat: 0.745744 recall: 0.834607 recall75: 0.160584 count: 10
I0724 11:03:10.931419 11724 yolov3_layer.cpp:411] noobj: 0.000832489 obj: 0.0108872 iou: 0.465499 cat: 0.753181 rec

**Comment by liwanning** [comment]
@eric612 
Model is YOLOv3-mobilenet-lite 
dataset is coco concert VOC
classes :4 . they are person car bus and truck.  
anchers was recalc  .
this is my config  files:
[mobilenet_yolov3_lite_train.txt](https://github.com/eric612/MobileNet-YOLO/files/3424877/mobilenet_yolov3_lite_train.txt)
[mobilenet_yolov3_lite_test.txt](https://github.com/eric612/MobileNet-YOLO/files/3424878/mobilenet_yolov3_lite_test.txt)
[mobilenet_yolov3_lite_solver.txt](https://github.com/eric612/MobileNet-YOLO/files/3424879/mobilenet_yolov3_lite_solver.txt)
[mobilenet_yolov3_lite_deploy_trt.txt](https://github.com/eric612/MobileNet-YOLO/files/3424880/mobilenet_yolov3_lite_deploy_trt.txt)

**Comment by liwanning** [comment]
@eric612 
How should i understand "noobj" and "obj" .I watch src/layer/yolov3_detection_output_layer .Do you mean they are scores of background and object ?

**Comment by jinfagang** [comment]
@liwanning Did u training from scratch? It's hard to converges if you training from start.
